src/Integration.Vsix.UnitTests/CFamily/CMake/CompilationDatabaseRequestFactoryTests.cs

```diff
@@ -57,11 +58,33 @@ public void TryGet_NoConfig_ReturnsNull()
             rulesConfigProvider.Invocations.Count.Should().Be(0);
         }
 
+        [TestMethod]
+        [Description("Check support for header files")]
+        public void TryGet_LanguageCalculatedBasedOnCompilationEntry()
+        {
+            const string fileName = "c:\\file.h";
+
+            var compilationDatabaseEntry = CreateCompilationDatabaseEntry("file.c");
+            var compilationConfigProvider = CreateCompilationProvider(fileName, compilationDatabaseEntry);
+            var rulesConfigProvider = new Mock<ICFamilyRulesConfigProvider>();
+
+            var testSubject = CreateTestSubject(compilationConfigProvider.Object, rulesConfigProvider.Object);
+            testSubject.TryGet(fileName, new CFamilyAnalyzerOptions());
+
+            compilationConfigProvider.VerifyAll();
+
+            // When analyzing header files, the analyzed file will be ".h", which is not a known rules' language.
+            // However, the compilation entry is a ".c" file, so we expect the code to calculate the rules based on the entry.
+            rulesConfigProvider.Verify(x=> x.GetRulesConfiguration(SonarLanguageKeys.C), Times.Once());
+        }
+
         [TestMethod]
         public void TryGet_UnrecognizedLanguage_ReturnsNull()
         {
             const string fileName = "c:\\file.txt";
-            var compilationConfigProvider = CreateCompilationProvider(fileName, new CompilationDatabaseEntry());
+
```

Comment:
Changed existing tests to create a valid CreateCompilationDatabaseEntry (with a file name)